services/basket/src/main/java/org/cryptimeleon/incentive/services/basket/storage/ItemInBasketEntity.java

```diff
@@ -0,0 +1,75 @@
+package org.cryptimeleon.incentive.services.basket.storage;
+
+import javax.persistence.*;
+import java.util.Objects;
+
+@Entity
+@Table(name = "item_in_basket")
+public class ItemInBasketEntity {
+    @EmbeddedId
+    private ItemInBasketId id;
+
+    @ManyToOne
+    @MapsId("basketId")
+    @JoinColumn(name = "basket_id")
+    private BasketEntity basket;
+
+    @ManyToOne
+    @MapsId("itemId")
+    @JoinColumn(name = "item_id")
+    private ItemEntity item;
+    private int count;
+
```

Comment:
I actually do not yet see much of the structure we discussed yesterday (Discord #work 10.11.22) in this class.
I guess this is because this will be implemented in the second step of refactoring after I merged feature/store-frontend to develop?

Reply:
Yes, as you can see in the BasketController, these classes will be converted into instances of the public API.